File: Server/src/ModManager.ts
import { DependencyContainer } from "tsyringe"

import { IPreSptLoadMod } from "@spt/models/external/IPreSptLoadMod"
import { IPostDBLoadMod } from "@spt/models/external/IPostDBLoadMod"
import { IPostSptLoadMod } from "@spt/models/external/IPostSptLoadMod"

import { AbstractModManager } from "./AbstractModManager";

import { AmmoManager } from "./AmmoManager"
import { GlobalsManager } from "./GlobalsManager"
import { HealingManager } from "./HealingManager"
import { HealthManager } from "./HealthManager"
import { MagazinesManager } from "./MagazinesManager"
import { MeleeManager } from "./MeleeManager"
import { RecoilManager } from "./RecoilManager"
import { FoodManager } from "./FoodManager"
import { StimulatorsManager } from "./StimulatorsManager"
import { ArmorManager } from "./ArmorManager"
import { WeaponsManager } from "./WeaponsManager"
import { EffectsManager } from "./EffectsManager"

export class ModManager implements IPreSptLoadMod, IPostDBLoadMod, IPostSptLoadMod
{
    private managers: AbstractModManager[] = []

    constructor()
    {
        this.managers.push(new AmmoManager)
        this.managers.push(new GlobalsManager)
        this.managers.push(new HealingManager)
        this.managers.push(new HealthManager)
        this.managers.push(new MagazinesManager)
        this.managers.push(new MeleeManager)
        this.managers.push(new RecoilManager)
        this.managers.push(new FoodManager)
        this.managers.push(new StimulatorsManager)
        this.managers.push(new ArmorManager)
        this.managers.push(new WeaponsManager)
        this.managers.push(new EffectsManager)
    }

    public preSptLoad(container: DependencyContainer): void
    {
        this.managers.forEach(m => m.preSptLoad(container))
    }
  
    public postDBLoad(container: DependencyContainer): void 
    {
        this.managers.forEach(m => m.postDBLoad(container))
    }

    public postSptLoad(container: DependencyContainer): void 
    {
        this.managers.forEach(m => m.postSptLoad(container))
    }
}

module.exports = { mod: new ModManager() }

File: Server/VAI-NRTP/ModManager.cs
using System.Diagnostics;
using SPTarkov.DI.Annotations;
using VAI.NRTP.Managers;

namespace VAI.NRTP;

[Injectable(InjectionType.Singleton)]
public sealed class ModManager
{
    private readonly List<AbstractModManager> _managers = [];

    public ModManager(
        AmmoManager ammoManager,
        GlobalsManager globalsManager,
        HealingManager healingManager,
        HealthManager healthManager,
        MagazinesManager magazinesManager,
        MeleeManager meleeManager,
        RecoilManager recoilManager,
        FoodManager foodManager,
        StimulatorsManager stimulatorsManager,
        ArmorManager armorManager,
        WeaponsManager weaponsManager,
        EffectsManager effectsManager,
        ModContext _)
    {
        _managers.Add(ammoManager);
        _managers.Add(globalsManager);
        _managers.Add(healingManager);
        _managers.Add(healthManager);
        _managers.Add(magazinesManager);
        _managers.Add(meleeManager);
        _managers.Add(recoilManager);
        _managers.Add(foodManager);
        _managers.Add(stimulatorsManager);
        _managers.Add(armorManager);
        _managers.Add(weaponsManager);
        _managers.Add(effectsManager);

        _managers.Sort((a, b) => a.Priority.CompareTo(b.Priority));
    }

    public void PreSptLoad()
    {
        foreach (var manager in _managers)
        {
            manager.PreSptLoad();
        }
    }

    public void PostDbLoad()
    {
        foreach (var manager in _managers)
        {
#if DEBUG
            var managerName = manager.GetType().Name;
            Constants.GetLogger().Info($"{Constants.ModTitle}: PostDB start {managerName}");
            var sw = Stopwatch.StartNew();
#endif
            manager.PostDbLoad();
#if DEBUG
            sw.Stop();
            Constants.GetLogger().Info($"{Constants.ModTitle}: PostDB end {managerName} ({sw.ElapsedMilliseconds}ms)");
#endif
        }
    }

    public void PostSptLoad()
    {
        foreach (var manager in _managers)
        {
            manager.PostSptLoad();
        }
    }

    public void FinalLoad()
    {
        foreach (var manager in _managers)
        {
            manager.FinalLoad();
        }
    }
}
